### ISSUE TITLE
<p> inside a <label> (not allowed) is in bootstrap3 templates

### ISSUE THREAD

**Issue by rrauenza** [django-crispy-forms/django-crispy-forms#632]
I'm validating my html for my website and I've found a case where crispy is emitting something technically invalid (this is output from my html5 validator):

```
CRITICAL: >>>>         <p id="hint_id_archived" class="help-block">blah blah blah</p> </label> </div> </div> </div> <div id="div_id_default_baseline" class="form-group"> <label for="id_default_baseline" class="control-label col-lg-2">
CRITICAL: >>>>          ^                                         ^
CRITICAL: error: Element “p” not allowed as child of element “label” in this context. (Suppressing further errors from this subtree.)

```

![image](https://cloud.githubusercontent.com/assets/1376251/18652713/7c31bc0a-7e89-11e6-94cb-be4b55e3906d.png)

I'm having a hard time searching through the bootstrap3 crispy templates and finding exactly where it is happening ... 

CRISPY_TEMPLATE_PACK = 'bootstrap3'

I'm using 1.6.0

It isn't a huge deal as the page is perfectly viewable, it's just strictly incorrect.

**Comment by carltongibson** [comment]
@rrauenza Thanks for the report. If you can track down the exact point of error that would be great.

**Comment by rrauenza** [comment]
Changing it to 

````
 <label for="{{ field.id_for_label }}" class="{% if field.field.required %} requiredField{% endif %}">
    {% crispy_field field %}                                    
    {{ field.label|safe }}                                      
</label>                                                        
{% include 'bootstrap3/layout/help_text_and_errors.html' %}     
````

...does seem to work, and I would argue renders better as well.  It is left justified now more consistently with other help text.

**Comment by rrauenza** [comment]
https://github.com/django-crispy-forms/django-crispy-forms/pull/633

**Comment by rrauenza** [comment]
Is there any chance of getting this pulled into a 1.6.1?

**Comment by zoidbergwill** [comment]
Thanks @rrauenza 

@carltongibson mentioned [this morning](https://github.com/django-crispy-forms/django-crispy-forms/pull/607#issuecomment-248216871) that'll be releasing a new version soon.

>In the meantime you can use pip to install straight from GitHub, using the [VCS Support features](https://pip.pypa.io/en/latest/reference/pip_install/#vcs-support).
>
>You'd need something like this in your requirements file:
>
>```
>-e git+https://github.com/django-crispy-forms/django-crispy-forms@dev#egg=django-crispy-forms
>```

**Comment by rrauenza** [comment]
Thanks!! -- I just found I can do that in requirements.txt as well and am just pointing to a revision on my fork for now.

I see you guys are also working on a test framework -- in case you want to validate your fragments, I've been investigating various html5 validators and narrowed down my choices either to https://html5.validator.nu/, which either can be run from a jar (slow, but simple, has optional json output), a local webservice using the jar, or using the online REST service.  Alternatively there is also a nodejs based mozilla validator.

I went with validator.nu locally as a jar.
